### PR TITLE
feat: redo rn 73 support

### PR DESCRIPTION
**Pull request by gosha212**

### CODE REVIEW

File: lib/android/app/build.gradle [hunk 1]
```diff
@@ -66,7 +66,11 @@ android {
         targetCompatibility JavaVersion.VERSION_1_8
     }
     kotlinOptions {
-        jvmTarget = JavaVersion.VERSION_1_8
+        if (reactNativeMinorVersion() >= 73) {
+            jvmTarget = JavaVersion.VERSION_17
+        } else {
+            jvmTarget = JavaVersion.VERSION_11
+        }
```
Comment (d4vidi) [inline lib/android/app/build.gradle:73]: ok, cool. Did you somehow check that it really does work with RN 71?

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/topbar/button/ButtonPresenter.kt [hunk 2]
```diff
@@ -194,8 +194,8 @@ open class ButtonPresenter(private val context: Context, private val button: But
 
                 class WixAccessibilityDelegateCompat: AccessibilityDelegateCompat(){
                     override fun onInitializeAccessibilityNodeInfo(
-                        host: View?,
-                        info: AccessibilityNodeInfoCompat?
+                        host: View,
+                        info: AccessibilityNodeInfoCompat
```
Comment (d4vidi) [inline lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/topbar/button/ButtonPresenter.kt:198]: Changes in SDK 34?

File: package.json [hunk 3]
```diff
@@ -76,7 +76,10 @@
     "@babel/plugin-proposal-export-namespace-from": "7.10.1",
     "@babel/runtime": "7.22.6",
     "@babel/types": "7.22.5",
-    "@react-native/metro-config": "^0.73.0",
+    "@babel/preset-env": "^7.22.9",
+    "@react-native/metro-config": "^0.73.2",
+    "@react-native/babel-preset": "^0.73.18",
+    "@react-native/typescript-config": "^0.73.1",
```
Comment (d4vidi) [inline package.json:82]: Is this really needed? Please try to remove

File: package.json [hunk 4]
```diff
@@ -94,7 +97,7 @@
     "@typescript-eslint/parser": "4.33.0",
     "babel-jest": "^27.0.0",
     "clang-format": "^1.4.0",
-    "detox": "20.18.3",
+    "detox": "20.19.5",
```
Comment (d4vidi) [inline package.json:100]: 20.20.0? 😄

File: playground/android/app/build.gradle [hunk 5]
```diff
@@ -13,9 +13,11 @@ def jscFlavor = 'org.webkit:android-jsc:+'
 android {
     compileSdkVersion rootProject.ext.get("compileSdkVersion")
 
+    namespace "com.reactnativenavigation.playground"
+
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
+        targetCompatibility JavaVersion.VERSION_17
```
Comment (d4vidi) [inline playground/android/app/build.gradle:20]: This won't prevent us from running the playground over RN71 - or would it?

File: playground/ios/scripts/pre-install.js [hunk 6]
```diff
@@ -1,33 +0,0 @@
-const fs = require('fs');
```
Comment (d4vidi) [inline playground/ios/scripts/pre-install.js:1]: Shouldn't this come inside an "if RN < 73"?

File: lib/android/app/build.gradle (hunk 1 above)
Comment (gosha212) [reply lib/android/app/build.gradle:73]: Yes

File: lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/topbar/button/ButtonPresenter.kt (hunk 2 above)
Comment (gosha212) [reply lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/stack/topbar/button/ButtonPresenter.kt:198]: Yes

**Review by d4vidi — APPROVED** [review]

**Comment by gosha212** [comment]
> not able to link by command npx rnn-link my project is with latest version of react native 0.73.6 and android is setup with Kotlin and don't no how to manually link the with .kt file in documentation manual link is only shown for java not of .kt file

@firasath-virtus 
Here a ticket. You are welcome to contribute. We have no capacity for this task in the near future unfortunately. 
https://github.com/wix/react-native-navigation/issues/7821

**Pull request merged**